### ISSUE TITLE
@classmethod does not count as ClassVar on a protocol

### ISSUE THREAD

**Issue by matejcik** [microsoft/pyright#2848]
this is with pyright **1.1.206**

the following code does not typecheck:
```python
from typing import Protocol

class P(Protocol):
    @classmethod
    def foo(cls) -> None:
        ...

class A:
    @classmethod
    def foo(cls) -> None:
        pass

def consumes_p(p: type[P]) -> None:
    p.foo()  # error: Protocol class "P" does not define "foo" as a ClassVar

consumes_p(A)
```
same happens with `@staticmethod`.

Classmethods and staticmethods should also count as allowable on `type[P]`.

**Comment by matejcik** [comment]
PEP-544 doesn't seem to mention classmethods at all, so this might not strictly be a bug. But if protocols aren't supposed to support @classmethods, it would be great if the protocol definition raised some sort of warning.

For the record, mypy supports this just fine.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a regression I recently introduced. It will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.207, which I just published. It will also be included in the next release of pylance.
